ios/brave-ios/Sources/BraveShared/Extensions/URLExtensions.swift

```diff
@@ -136,6 +136,56 @@ extension URL {
 
     return URL(string: "\(baseURL)?\(InternalURL.Param.url.rawValue)=\(encodedURL)")
   }
+
+  public var hasMixedBidiDirection: Bool {
+    // First format the URL which will decode the puny-coding
+    let scheme = scheme ?? "http"
+    var renderedString = URLFormatter.formatURL(
+      absoluteString,
+      formatTypes: [.omitDefaults, .omitTrivialSubdomains, .omitTrailingSlashOnBareHostname],
+      unescapeOptions: .normal
+    )
+
+    // Strip prefixes
+    if let range = renderedString.range(of: "^(www|mobile|m)\\.", options: .regularExpression) {
+      renderedString.replaceSubrange(range, with: "")
+    }
+
+    // Strip scheme
+    if let range = renderedString.range(
+      of: "^(\(scheme)://|\(scheme):)",
+      options: .regularExpression
+    ) {
+      renderedString.replaceSubrange(range, with: "")
+    }
+
+    return renderedString.bidiDirection == .MIXED
+  }
+
+  public var isLTRRendered: Bool {
+    // First format the URL which will decode the puny-coding
+    let scheme = scheme ?? "http"
+    var renderedString = URLFormatter.formatURL(
+      absoluteString,
+      formatTypes: [.omitDefaults, .omitTrivialSubdomains, .omitTrailingSlashOnBareHostname],
+      unescapeOptions: .normal
+    )
+
+    // Strip prefixes
+    if let range = renderedString.range(of: "^(www|mobile|m)\\.", options: .regularExpression) {
```

Comment:
doesn't `omitTrivialSubdomains` above do the same?

Reply:
Yeah it does. I'll remove this.